src/ui/roles/common/RoleForm.js

```diff
@@ -27,154 +25,198 @@ const msgs = defineMessages({
   },
 });
 
-export class RoleFormBody extends Component {
-  constructor(props) {
-    super(props);
-    this.handleToggleSuperuser = this.handleToggleSuperuser.bind(this);
-  }
-  componentWillMount() {
-    this.props.onWillMount(this.props);
-  }
-
-  onSubmit = (ev) => {
-    ev.preventDefault();
-    this.props.handleSubmit();
-  };
+export const RoleFormBody = ({
+  onWillMount,
+  submitForm,
+  permissions,
+  onDiscard,
+  onCancel,
+  intl,
+  dirty,
+  isValid,
+  mode,
+  loading,
+  roleCode,
+  isSubmitting,
+  values,
+  setFieldValue,
+  handleChange,
+}) => {
+  const invalid = !isValid;
+  const isEdit = mode === EDIT_MODE;
+  const superuserToggled = values.permissions ? values.permissions.superuser : false;
 
-  handleToggleSuperuser(superuserToggled) {
-    const { onToggleSuperuser, permissions } = this.props;
-    onToggleSuperuser({ superuserToggled, permissions });
-  }
+  useEffect(() => {
+    onWillMount(roleCode);
+  }, [roleCode, onWillMount]);
```

Comment:
This is just trivial but I think the `onWillMount` naming seems to refer more closely to the deprecated lifecycle method. Perhaps a better one would be `onDidMount` or simply just `onMount`?